### DISCUSSION TITLE
Documentation proposal

### DISCUSSION THREAD

**Discussion by doanguyen** [simple-login/app#391] [Feature Requests]
I am trying to work on the docker image on arm64 and I see the documentation is a bit long and difficult to navigate. Do you think it's a good idea to create a static documentation site, such as docs.simplelogin.io 

I can work on that in the following days

**Comment by nguyenkims** [comment]
The current https://docs.simplelogin.io/ is used for the documentation to integrate the `Sign in with SimpleLogin` button. 

I think we can shorten the doc by extracting the API and OAuth sections to separated pages.

For the self-hosting part (which is the longest section), I think it'd be nice to create a single script (in bash?) to run all steps, something like:

```bash
./setup-simplelogin.sh --email-domain=mydomain.com --app-domain=app.mydomain.com --db-password=my-secret-password
```

that runs all the steps and gives users DNS instructions at the end.

For example, https://mailinabox.email/ has a script like this.

**Comment by FarisZR** [comment]
> The current https://docs.simplelogin.io/ is used for the documentation to integrate the `Sign in with SimpleLogin` button.
> 
> I think we can shorten the doc by extracting the API and OAuth sections to separated pages.
> 
> For the self-hosting part (which is the longest section), I think it'd be nice to create a single script (in bash?) to run all steps, something like:
> 
> ```shell
> ./setup-simplelogin.sh --email-domain=mydomain.com --app-domain=app.mydomain.com --db-password=my-secret-password
> ```
> 
> that runs all the steps and gives users DNS instructions at the end.
> 
> For example, https://mailinabox.email/ has a script like this.

i would recommend using something like mkdocs with mkdocs-material theme to create a static doc site on gh-pages, you can list all the docs there with every detail its better than using the readme file.

i have personally used it with my main project([arhackintosh](https://github.com/arhackintosh)) and it works really good!, i can help with setting up github actions CI and the repo in general
